### DISCUSSION TITLE
Issue with GetCatalogItemAsync when passing ASIN as input in catalog-items-api-v2020-12-01 and catalog-items-api-v2022-04-01

### DISCUSSION THREAD

**Discussion by SanketSatwase** [amzn/selling-partner-api-models#3640] [General] [answered]
I'm facing an issue while using the GetCatalogItemAsync method from the catalog-items-api-v2020-12-01 and catalog-items-api-v2022-04-01 in the Selling Partner API. The method takes a string ASIN (Amazon Standard Identification Number) as input, but when I provide an ASIN value, I receive the following ApiException:

**SellingPartnerAPI.catalogItems2022.Client.ApiException: 'Error converting value "B078H38SDP" to type 'SellingPartnerAPI.catalogItems2022.Model.ItemAsin'. Path 'asin', line 1, position 20.'**

I noticed that the error message suggests a problem with converting the provided ASIN value to the ItemAsin type. However, the GetCatalogItemAsync method expects a string as the ASIN parameter, not an ItemAsin object.

Here's a code snippet that reproduces the issue:
`private async Task<string> FetchItemDetails(string asin, string accessToken)
{
    try
    {
        List<string> marketplaceIds = new List<string> { "ATVPDKIKX0DER" };
        var getItemResponse = await catalogApi.GetCatalogItemAsync(asin, marketplaceIds);
        var item = getItemResponse;
        string itemDetails = JsonConvert.SerializeObject(item);
        return itemDetails;
    }
    catch (ApiException ex)
    {
        MessageBox.Show("An error occurred while fetching item details: " + ex.Message);
        return null;
    }
}`

Could you please help me understand why this error is occurring and how I can resolve it? I appreciate any insights or suggestions you may have. Thank you!

**Comment by vaidas-lungis** [comment]
Did you generate code from the provided JSON model? Maybe generator has difficulties and try with different one?

**Reply by SanketSatwase** [reply, marked as answer]
- I appreciate your suggestion to try using a different code generator. Yes, I generated the code from the provided JSON model. The code generation process seemed to work smoothly, and I followed the generated code's structure to make API calls.

- However, I encountered an issue specifically with the GetCatalogItemAsync method from the catalog-items-api-v2022-04-01 and catalog-items-api-v2020-12-01 version. Although the method expects a string ASIN parameter, I received an ApiException stating that there was a problem converting the ASIN value to the ItemAsin type. This puzzled me as the parameter should be a string, not an ItemAsin object.

- To resolve the issue, I manually modified the constructor of the Item class to accept a string parameter for the asin argument, instead of an ItemAsin object. This resolved the type mismatch error, and I can now successfully call the GetCatalogItemAsync method with a string ASIN value.

Here's the modification:

1. 

```
[DataMember(Name="asin", EmitDefaultValue=false)]
public ItemAsin Asin { get; set; }
```
As
```
[DataMember(Name = "asin", EmitDefaultValue = false)]
public string Asin { get; set; }
```

2. 

`public Item(ItemAsin asin = default(ItemAsin)`
As
`public Item(string asin = null`

**Reply by diegocvazquez** [reply]
@SanketSatwase 
Really good catch! was stucked in the same issue!
